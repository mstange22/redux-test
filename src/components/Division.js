import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Division = ({ a, b }) => (
  <div className="division-container">
    <h2>Division</h2>
    <h2>{a}</h2>
    <h2>/</h2>
    <h2>{b}</h2>
    <h2>=</h2>
    <h2>{b === 0 ? Infinity : a / b}</h2>
  </div>
);

Division.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  a: parseInt(state.a, 10),
  b: parseInt(state.b, 10),
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Division);
