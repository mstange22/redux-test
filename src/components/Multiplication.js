import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Multiplication = ({ a, b }) => (
  <div className="multiplication-container">
    <h2>Multiplication</h2>
    <h2>{a}</h2>
    <h2>*</h2>
    <h2>{b}</h2>
    <h2>=</h2>
    <h2>{a * b}</h2>
  </div>
);

Multiplication.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  a: parseInt(state.a, 10),
  b: parseInt(state.b, 10),
});

const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(Multiplication);
