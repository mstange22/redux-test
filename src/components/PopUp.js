import React from 'react';
import PropTypes from 'prop-types';

const PopUp = ({ title, message }) => (
  <div
    style={{
      textAlign: 'center',
      height: 100,
      width: 400,
      position:
      'fixed',
      top: 200,
      left: '50%',
      marginLeft: -220,
      backgroundColor: '#f4f442',
      border: 'solid 1px black',
      borderRadius: 10,
      display: 'flex',
      justifyContent: 'flex-start',
      padding: 20,
      flexDirection: 'column',
    }}
  >
    <div className="popup-title" style={{ fontWeight: 700 }}>
      {`Pop-Up Says "${title}"`}
    </div>
    <br />
    <div className="popup-message">
      {message}
    </div>
  </div>
);

PopUp.propTypes = {
  title: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
};

export default PopUp;
