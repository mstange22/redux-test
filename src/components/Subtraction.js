import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Subtraction = ({ a, b }) => (
  <div className="subtraction-container">
    <h2>Subtraction</h2>
    <h2>{a}</h2>
    <h2>-</h2>
    <h2>{b}</h2>
    <h2>=</h2>
    <h2>{a - b}</h2>
  </div>
);

Subtraction.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.number.isRequired,
};

const mapStateToProps = state => ({
  a: parseInt(state.a, 10),
  b: parseInt(state.b, 10),
});

const mapDispatchToProps = {
};

export default connect(mapStateToProps, mapDispatchToProps)(Subtraction);
