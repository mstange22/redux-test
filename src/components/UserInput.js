import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setValueA, setValueB } from '../redux/reducer';

const UserInput = (props) => {
  const onChange = (e) => {
    const { name, value } = e.target;
    if (/^\d+$/.test(value) || value === '') {
      if (name === 'a') props.setValueA(value === '' ? 0 : value);
      else props.setValueB(value === '' ? 0 : value);
    }
  };

  return (
    <div className="user-input-container">
      <b>Redux Calculator</b>
      <br />
      Enter two values
      <div className="input-container">
        <input
          className="user-input"
          name="a"
          onChange={onChange}
          value={props.a}
          autoComplete="off"
          autoFocus
        />
        <br />
        <input
          className="user-input"
          name="b"
          onChange={onChange}
          value={props.b}
          autoComplete="off"
        />
      </div>
    </div>
  );
};

UserInput.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.number.isRequired,
  setValueA: PropTypes.func.isRequired,
  setValueB: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  a: parseInt(state.a, 10),
  b: parseInt(state.b, 10),
});

const mapDispatchToProps = {
  setValueA,
  setValueB,
};

export default connect(mapStateToProps, mapDispatchToProps)(UserInput);
