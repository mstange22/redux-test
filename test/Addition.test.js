import React from 'react';
import renderer from 'react-test-renderer';
import { Addition } from '../src/components/Addition';

test('This is a test', () => {
  const props = { a: 0, b: 0 };
  const component = renderer.create(
    <Addition {...props} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
