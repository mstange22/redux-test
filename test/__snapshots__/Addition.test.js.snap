// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is a test 1`] = `
<div
  className="addition-container"
>
  <h2>
    Addition
  </h2>
  <h2>
    0
  </h2>
  <h2>
    +
  </h2>
  <h2>
    0
  </h2>
  <h2>
    =
  </h2>
  <h2>
    0
  </h2>
</div>
`;
